r/src/compute.cpp

```diff
@@ -223,4 +205,66 @@ std::shared_ptr<arrow::Table> Table__FilterChunked(
   }
   return tab;
 }
+
+arrow::Datum to_datum(SEXP x) {
```

Comment:
We chose not to expose the Datum class in R at all. Datum is redundant because we already have the SEXP box for different inputs, so we can just deal with that in the interface. 

Unrelated, see comment on the next line.

Reply:
I also don't think it needs to be exposed

Reply:
How's this:
```c++
template <typename T>
std::shared_ptr<T> MaybeUnbox(const char* class_name, SEXP x) {
  if (Rf_inherits(x, "ArrowObject") && Rf_inherits(x, class_name)) {
    Rcpp::ConstReferenceSmartPtrInputParameter<std::shared_ptr<T>> obj(x);
    return static_cast<std::shared_ptr<T>>(obj);
  }
  return nullptr;
}

arrow::Datum to_datum(SEXP x) {
  if (auto array = MaybeUnbox<arrow::Array>("Array", x)) {
    return array;
  }

  if (auto chunked_array = MaybeUnbox<arrow::ChunkedArray>("ChunkedArray", x)) {
    return chunked_array;
  }

  // ...
```